Dockerfile

```diff
@@ -2,12 +2,18 @@
 # Use of this source code is governed by an Apache2
 # license that can be found in the LICENSE file.
 
-FROM scratch
+FROM alpine:latest
```

Comment:
Ah, OK, the reason it's failing is because /tmp doesn't exist in the scratch image. It should be fairly easy to work around this (either create it at build time, or modify OPA to create the directory on startup if it doesn't exist). That being said, logging to stderr seems like the way to go, e.g., no need to do rotation, worry about disk overflowing, can easily hook up to syslog, etc.

Right now there wouldn't be much gained by having a shell inside the container--I know what you're describing, and it's definitely valuable for heavy-weight images, but in this case, I haven't found a need.

Based on this, I'm not sure we gain much. Maybe it's easier to support TLS?